web/libs/ui/src/lib/sparkles/sparkles.tsx

```diff
@@ -0,0 +1,510 @@
+import React from "react";
+import { IconSparkle } from "@humansignal/icons";
+import { usePrefersReducedMotion, useRandomInterval } from "../../../hooks/hooks";
+import styles from "./sparkles.module.scss";
+import clsx from "clsx";
+
+/**
+ * Options for randomPositionAvoidingCenter
+ */
+export interface SparkleAreaOptions {
+  areaShape: "circle" | "rect";
+  areaWidth?: number;
+  areaHeight?: number;
+  areaRadius?: number;
+  cutoutShape: "circle" | "rect";
+  cutoutWidth?: number;
+  cutoutHeight?: number;
+  cutoutRadius?: number;
+  center: { x: number; y: number };
+}
+
+/**
+ * Interface for a Sparkle object
+ */
+export interface Sparkle {
+  id: string;
+  createdAt: number;
+  color: string;
+  size: number;
+  style: {
+    position: "absolute";
+    top: string;
+    left: string;
+    pointerEvents: "none";
+    zIndex: number;
+  };
+}
```

Comment:
Since there a lot of types, maybe we move these to a types.ts file in this same sparkles folder to keep the component file nice and lean.